soccerdata/sofascore.py

```diff
@@ -0,0 +1,265 @@
+             If True, will force the use of cached data anyway.
+
+        Returns
+        -------
+        pd.DataFrame
+        """
+        filemask = 'seasons/{}_{}.html'
+        urlmask = SOFASCORE_API + "unique-tournament/{}/season/{}/standings/total"
+
+        idx = ['league', 'season']
+        cols = ['team', 'MP', 'W', 'D', 'L', 'GF', 'GA', 'GD', 'Pts']
+
+        seasons = self.read_seasons()
+        # collect league tables
+        mult_tables = []
+        for (lkey, skey), season in seasons.iterrows():
+            filepath = self.data_dir / filemask.format(lkey, skey)
+            url = urlmask.format(season.league_id, season.season_id)
+            current_season = not self._is_complete(lkey, skey)
+            reader = self.get(url, filepath, no_cache=current_season and not force_cache)
+            season_data = json.load(reader)
+            for row in season_data['standings'][0]['rows']:
+                mult_tables.append(
+                    {
+                        'league': lkey,
+                        'season': skey,
+                        'team': row['team']['name'],
+                        'MP': row['matches'],
+                        'W': row['wins'],
+                        'D': row['draws'],
+                        'L': row['losses'],
+                        'GF': row['scoresFor'],
+                        'GA': row['scoresAgainst'],
+                        'GD': row['scoresFor'] - row['scoresAgainst'],
+                        'Pts': row['points'],
+                    }
+                )
+            df = (
+                pd.DataFrame(mult_tables)
+                .set_index(idx)
+                .replace({'team': TEAMNAME_REPLACEMENTS})
+                .sort_index()[cols]
+            )
+        return df
+
+    def read_schedule(self) -> pd.DataFrame:
+        """Retrieve the game schedule for the selected leagues and seasons.
+
+        Returns
+        -------
+        pd.DataFrame
+        """
+        urlmask1 = SOFASCORE_API + "unique-tournament/{}/season/{}/rounds"
+        urlmask2 = SOFASCORE_API + "unique-tournament/{}/season/{}/events/round/{}"
+
+        cols = [
+            'round',
+            'week',
+            'date',
+            'home_team',
+            'away_team',
+            'home_score',
+            'away_score',
+            'game_id',
+        ]
+
+        df_seasons = self.read_seasons()
+        all_schedules = []
+        for (lkey, skey), season in df_seasons.iterrows():
+            # filepath = self.data_dir / filemask.format(lkey, skey)
+            url1 = urlmask1.format(season['league_id'], season['season_id'])
+            reader1 = self.get(url1)
+            season_data = json.load(reader1)
+            rounds = season_data['rounds']
+
+            for round in rounds:
+                url2 = urlmask2.format(season['league_id'], season['season_id'], round['round'])
+                reader2 = self.get(url2)
```

Comment:
Why do you not use caching here?

Reply:
I have implemented caching now. Apologies for that